### PR TITLE
allow non-dev server usage that is not just heroku

### PR DESCRIPTION
**Pull request by jhudson8**
Add support for a `--standalone` command line argument.  This would be used to indicate that the server is not running as a developer proxy server.

The heroku deployments would automatically get the standalone attribute set so there is no need to supply that argument for heroku.

### CODE REVIEW

File: lib/config-service.js [hunk 1]
```diff
@@ -12,8 +12,8 @@ module.exports = function(req, res, next) {
   var serverConfig = config.serverConfig.config;
 
   var portConfig = {
-    port: config.isHeroku ? 80 : config.port,
-    securePort: config.isHeroku ? 443 : config.securePort
+    port: config.isStandalone ? 80 : config.port,
+    securePort: config.isStandalone ? 443 : config.securePort
```
Comment (kpdecker) [inline lib/config-service.js]: This is a case that we need isHeroku, per our conversation earlier. Basically we need to tell the client to hit a different port than we are actually serving from as heroku implements a proxy.

File: lib/index.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ var app = express.createServer();
 configure(app, false);
 app.listen(config.port);
 
-if (!config.isHeroku) {
+if (!config.isStandalone) {
   console.log('Starting mock-server on secure port', config.securePort);
   try {
     var hostExists = fs.statSync(config.appDir + '/host.key').isFile();
```
Comment (kpdecker) [inline lib/index.js:77]: This is unrelated but you might run into issues with ssl keys when you get to that point. Just FYI and can be handled in future PR.

File: lib/config-service.js (hunk 1 above)
Comment (jhudson8) [inline lib/config-service.js]: ah ok thanks

**Comment by jhudson8** [comment]
@patrickkettner yes and thanks - I didn't actually realize that we could have non-project .gitignore files.  will do

File: lib/config-service.js (hunk 1 above)
Comment (kpdecker) [inline lib/config-service.js]: Alternatively we could do some sort of `isProxied` vs. `isHeroku` which might make the intention a bit clearer.

**Comment by kpdecker** [comment]
@patrickkettner death to non-vim users!!!!!! ;)

**Comment by kpdecker** [comment]
Actually on a more serious note I'd prefer that we keep ignores inside of the repo so we have uniform behavior across all users regardless of their *ahem\* editor choice. In this case it probably doesn't matter as .project is unlikely to have name conflicts but I don't want us to have too many variations in what our tools actually do.

**Comment by kpdecker** [comment]
Joe this looks fine to me to merge.

**Pull request merged**